src/Order/Checkout/CheckoutForm.php

```diff
@@ -0,0 +1,269 @@
+<?php
+declare(strict_types=1);
+
+namespace SwipeStripe\Order\Checkout;
+
+use SilverStripe\Control\Controller;
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Control\RequestHandler;
+use SilverStripe\Forms\EmailField;
+use SilverStripe\Forms\FieldList;
+use SilverStripe\Forms\Form;
+use SilverStripe\Forms\FormAction;
+use SilverStripe\Forms\HiddenField;
+use SilverStripe\Forms\OptionsetField;
+use SilverStripe\Forms\TextField;
+use SilverStripe\Omnipay\GatewayFieldsFactory;
+use SilverStripe\Omnipay\GatewayInfo;
+use SilverStripe\Omnipay\Model\Message\CompletePurchaseError;
+use SilverStripe\Omnipay\Model\Message\PaymentMessage;
+use SilverStripe\Omnipay\Model\Message\PurchaseError;
+use SilverStripe\Omnipay\Model\Payment;
+use SilverStripe\Omnipay\Service\ServiceFactory;
+use SwipeStripe\Order\Order;
+use SwipeStripe\Order\OrderConfirmationPage;
+use SwipeStripe\Order\PaymentExtension;
+use SwipeStripe\Order\PaymentStatus;
+use SwipeStripe\Price\SupportedCurrencies\SupportedCurrenciesInterface;
+
+/**
+ * Class CheckoutForm
+ * @package SwipeStripe\Order\Checkout
+ * @property Payment|null $PaymentError
+ */
+class CheckoutForm extends Form
+{
+    const ORDER_HASH_FIELD = 'OrderContents';
+    const PAYMENT_METHOD_FIELD = 'PaymentMethod';
+    const PAYMENT_ID_QUERY_PARAM = 'payment';
+
+    const CHECKOUT_GUEST = 'Guest';
+    const CHECKOUT_CREATE_ACCOUNT = 'Account';
+
+    /**
+     * @var array
+     */
+    private static $dependencies = [
+        'paymentServiceFactory' => '%$' . ServiceFactory::class,
+        'supportedCurrencies'   => '%$' . SupportedCurrenciesInterface::class,
+    ];
+
+    /**
+     * @var ServiceFactory
+     */
+    public $paymentServiceFactory;
```

Comment:
Please use `set<Service>(<ServiceType> $service)` / `get<Service>()` instead of public properties.